docs/adr-0003-discrete_metrics_table_structure.md

```diff
@@ -0,0 +1,103 @@
+
+
+## Decision Drivers
+
+* Flexible to individual metrics but also groups of metrics (i.e., by data source)
+* Flexible to adding/removing metrics later
+* Backwards compatibility
+* Complexity
+
+
+## Decision Outcome
+
+**Option 2** is the chosen option. We can preserve the schemas of the views by pivoting the results tables, and we save a lot on complexity during normal execution as compared to Option 1. Option 3 is very similar but creates additional clutter in the form of many tables.
+
+
+## Options
+
+For each option, we will use the following scenario as an example for describing how the option will work:
+
+* Experiment with 9 metrics, divided equally between 3 metric data sources
+* Metric are initially computed discretely by data source (i.e., 3 metrics queries, with 3 metrics computed in each query)
+* Sometime after enrollment ends and some results have been computed, one metric is removed, and a different new metric is added via a custom experiment config
+
+### 1. Dynamic Table Schema with MERGE
+
+* For each metric query:
+  * If table does not exist:
+    * Create with schema of query result
+  * If table exists:
+    * Create new columns for each metric
+    * Use MERGE statement to populate new columns in-place ([ref](https://cloud.google.com/bigquery/docs/reference/standard-sql/dml-syntax#merge_statement))
+* Metric is removed
+  * Compare metric columns to new set of metrics to determine that we have an extra one in the results
+  * For each analysis basis and analysis window:
+    * Drop column for metric
+    * Delete rows in statistics tables for metric
+* Metric is added
+  * Compare metric columns to new set of metrics to determine that we have a new one to compute
+  * For each analysis basis and analysis window:
+    * Create metric query for the new metric
+    * Same as original execution under "If table exists:"
+      * Create new column for new metric
+      * Use MERGE statement to populate new column in-place ([ref](https://cloud.google.com/bigquery/docs/reference/standard-sql/dml-syntax#merge_statement))
+    * Compute statistics
+
+#### Pros / Cons
+
+* **+** This preserves the existing table structure
+* **-** Lots of added complexity
+* **-** Added cost (BigQuery treats each MERGE as basically a full delete/recreate of the table)
+
+
+### 2. Row per Metric
```

Comment:
I don't _love_ this option, but of the ones we came up so far I think this might be the best one

Reply:
Yea I feel the same way. I guess the question is whether we're happy enough with this to proceed, or if we dislike them all enough to go back to the drawing board. I'd vote move forwards with this approach obviously, but open to discussion.